src/mongo/nodes.ts

```diff
@@ -257,7 +257,7 @@ export class MongoCollectionNode implements IEditableNode {
 	async getChildren(): Promise<INode[]> {
 		if (!this._hasFetched) {
 			this._children = [];
-			this._iterator = this.collection.find(this.query);
+			this._iterator = this.collection.find(JSON.parse(this.query));
```

Comment:
You did a lot of work to ensure `this.query` is `null` instead of `undefined`, but it would seem less bug-prone if you just did this:
```
this._iterator = this.query ? this.collection.find(JSON.parse(this.query)) : this.collection.find();
```

That way we don't have to worry if you 'missed' one case where it was supposed to be null instead of undefined